File: fixtures/scale/dart/large/lib/ciphers/foursquare/foursquare_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'foursquare_cipher.dart';
import 'foursquare_key.dart';
import 'foursquare_vectors.dart';

class FourSquareSuite implements CipherSuite {
  @override
  String name() => 'foursquare';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = FourSquareCipher(FourSquareKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in foursquareSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/java/large/src/cryptobench/ciphers/foursquare/FourSquareSuite.java
package cryptobench.ciphers.foursquare;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class FourSquareSuite implements CipherSuite {
    @Override
    public String name() {
        return "foursquare";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new FourSquareCipher(FourSquareKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : FourSquareVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}
